### ISSUE TITLE
Docker not starting

### ISSUE THREAD

**Issue by SantoDE** [docker/for-mac#50]
### Expected behavior
Docker works properly

### Actual behavior
Docker tries to start "forever" but nothing happens.

### Information

Diagnostic ID: C2193A9E-94BC-4376-BE22-93666DE1E973
Docker for Mac: 1.12.0-beta21 (Build 10868)
macOS: Version 10.11.6 (Build 15G31)
[ERROR]  docker-cli
         docker ps failed
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

1. Start Docker on Mac

**Comment by djs55** [comment]
`docker ps` hangs because the connection to `/var/run/docker.sock` is not working:
```
daemon.info vsudd:
 1 error copying from docker to vsock: read unix @->/var/run/docker.sock: read: connection reset by peer
```
The `docker.log` reveals that the docker engine failed to start due to AUFS errors:
```
time="2016-08-01T11:25:50.961276789Z" level=debug msg="[graphdriver] trying provided driver \"aufs\""
time="2016-08-01T11:25:50.965876938Z" level=debug msg="Using graph driver aufs"
time="2016-08-01T11:25:51.099387490Z" level=debug msg="Failed to load layer sha256:0378120e0a115277db6a303f56996c56ced8b235659a08635b
1b9ebeb1d3672e: failed to get diff id for sha256:0378120e0a115277db6a303f56996c56ced8b235659a08635b1b9ebeb1d3672e: invalid checksum d
igest format"
time="2016-08-01T11:25:51.195903729Z" level=debug msg="Failed to load layer sha256:3a388107152d0675cd9528cd5402ca335b358ac58caeb8029c
1ddd518e0e8555: failed to get diff id for sha256:3a388107152d0675cd9528cd5402ca335b358ac58caeb8029c1ddd518e0e8555: invalid checksum d
igest format"
time="2016-08-01T11:25:51.201566956Z" level=debug msg="Failed to load layer sha256:44bc0843942ebe22d99919335133b196a9575e6caf9176e111da3d6bcf2c230e: failed to get diff id for sha256:44bc0843942ebe22d99919335133b196a9575e6caf9176e111da3d6bcf2c230e: invalid checksum digest format"
time="2016-08-01T11:25:51.220756051Z" level=debug msg="Failed to load layer sha256:6db2356988c8db01a0ca074e8391c786a5718c4814afdadb3ec28983bc81c230: failed to get diff id for sha256:6db2356988c8db01a0ca074e8391c786a5718c4814afdadb3ec28983bc81c230: invalid checksum digest format"
time="2016-08-01T11:25:51.224542829Z" level=debug msg="Failed to load layer sha256:865473ec3a1e2c1089d6cccc83a20b0c85b146ce6a2d067ca8b3c95bb36986f9: failed to get diff id for sha256:865473ec3a1e2c1089d6cccc83a20b0c85b146ce6a2d067ca8b3c95bb36986f9: invalid checksum digest format"
time="2016-08-01T11:25:51.225559055Z" level=debug msg="Failed to load layer sha256:941ef2028037ee554da56e1307d378871991e76e76fd1c7aac1b9362b1a068db: failed to get diff id for sha256:941ef2028037ee554da56e1307d378871991e76e76fd1c7aac1b9362b1a068db: invalid checksum digest format"
time="2016-08-01T11:25:51.233156555Z" level=debug msg="Failed to load layer sha256:b9ef87c79c6fe428e26af74372878b73a4706b11b541916bd4b923ed4009bc78: failed to get diff id for sha256:b9ef87c79c6fe428e26af74372878b73a4706b11b541916bd4b923ed4009bc78: invalid checksum digest format"
time="2016-08-01T11:25:51.235869260Z" level=debug msg="Failed to load layer sha256:c160e74fb9565824f522a5211d7cf44e70d1652d06692ec3cc64390e8237ee85: failed to get diff id for sha256:c160e74fb9565824f522a5211d7cf44e70d1652d06692ec3cc64390e8237ee85: invalid checksum digest format"
time="2016-08-01T11:25:51.201566956Z" level=debug msg="Failed to load layer sha256:44bc0843942ebe22d99919335133b196a9575e6caf9176e111da3d6bcf2c230e: failed to get diff id for sha256:44bc0843942ebe22d99919335133b196a9575e6caf9176e111da3d6bcf2c230e: invalid checksum digest format"
time="2016-08-01T11:25:51.220756051Z" level=debug msg="Failed to load layer sha256:6db2356988c8db01a0ca074e8391c786a5718c4814afdadb3ec28983bc81c230: failed to get diff id for sha256:6db2356988c8db01a0ca074e8391c786a5718c4814afdadb3ec28983bc81c230: invalid checksum digest format"
time="2016-08-01T11:25:51.224542829Z" level=debug msg="Failed to load layer sha256:865473ec3a1e2c1089d6cccc83a20b0c85b146ce6a2d067ca8b3c95bb36986f9: failed to get diff id for sha256:865473ec3a1e2c1089d6cccc83a20b0c85b146ce6a2d067ca8b3c95bb36986f9: invalid checksum digest format"
time="2016-08-01T11:25:51.225559055Z" level=debug msg="Failed to load layer sha256:941ef2028037ee554da56e1307d378871991e76e76fd1c7aac1b9362b1a068db: failed to get diff id for sha256:941ef2028037ee554da56e1307d378871991e76e76fd1c7aac1b9362b1a068db: invalid checksum digest format"
time="2016-08-01T11:25:51.233156555Z" level=debug msg="Failed to load layer sha256:b9ef87c79c6fe428e26af74372878b73a4706b11b541916bd4b923ed4009bc78: failed to get diff id for sha256:b9ef87c79c6fe428e26af74372878b73a4706b11b541916bd4b923ed4009bc78: invalid checksum digest format"
time="2016-08-01T11:25:51.235869260Z" level=debug msg="Failed to load layer sha256:c160e74fb9565824f522a5211d7cf44e70d1652d06692ec3cc64390e8237ee85: failed to get diff id for sha256:c160e74fb9565824f522a5211d7cf44e70d1652d06692ec3cc64390e8237ee85: invalid checksum digest format"
time="2016-08-01T11:25:51.236326796Z" level=debug msg="Failed to load layer sha256:d4703a62d89dc203a82d0dcebbc21ca090f5958ca703f886f91d68e88a7f796b: failed to get diff id for sha256:d4703a62d89dc203a82d0dcebbc21ca090f5958ca703f886f91d68e88a7f796b: invalid checksum digest format"
time="2016-08-01T11:25:51.238411204Z" level=debug msg="Failed to load layer sha256:ebe5a55e5ce2949476b4be22aef5b8a1ebeecf7598626195043e9b1c8c7c2a28: failed to get diff id for sha256:ebe5a55e5ce2949476b4be22aef5b8a1ebeecf7598626195043e9b1c8c7c2a28: invalid checksum digest format"
time="2016-08-01T11:25:51.240457799Z" level=debug msg="Failed to load mount 2e75f1ba342aaf7371d8ec0d3b65c154f4fc3a0ed89100335734b0074df30054: invalid mount id value"
time="2016-08-01T11:25:51.240986186Z" level=debug msg="Failed to load mount 65aaf1597c6e8441f90a5b064b13863cd16dfc3f7a69c424783264e6e5972614: invalid mount id value"
time="2016-08-01T11:25:51.245154082Z" level=debug msg="Max Concurrent Downloads: 3"
time="2016-08-01T11:25:51.245459210Z" level=debug msg="Max Concurrent Uploads: 5"
time="2016-08-01T11:25:51.342079965Z" level=debug msg="Cleaning up old mountid : start."
time="2016-08-01T11:25:51.343174225Z" level=fatal msg="Error starting daemon: layer does not exist"
```

**Comment by djs55** [comment]
A workaround which should allow docker to start is "Whale menu -> Preferences -> Uninstall/Reset -> Reset to factory defaults". Note this will remove all images and containers, so they will need to be rebuilt :(

There have been lots of bug fixes in docker engine 1.12.1, which is already available on the beta channel. It's worth trying again with this later version.
